src/util/Rtree.cpp

```diff
@@ -0,0 +1,1277 @@
+//  Copyright 2023, University of Freiburg,
+//                  Chair of Algorithms and Data Structures.
+//  Author: Noah Nock <noah.v.nock@gmail.com>
+
+#include <util/Rtree.h>
+#include <util/BackgroundStxxlSorter.h>
+
+static void centerOrdering(multiBoxGeo& boxes, size_t dim) {
+  if (dim == 0) {
+    // order by centerX
+    sortRuleLambdaX comp;
+
+    std::sort(boxes.begin(), boxes.end(), comp);
+  } else {
+    // order by centerY
+    auto sortRuleLambda = [](rTreeValue b1, rTreeValue b2) -> bool {
+      double center1 = (b1.box.min_corner().get<1>() + b1.box.max_corner().get<1>()) / 2;
+      double center2 = (b2.box.min_corner().get<1>() + b2.box.max_corner().get<1>()) / 2;
+      return center1 < center2;
+    };
+
+    std::sort(boxes.begin(), boxes.end(), sortRuleLambda);
+  }
+}
+
+static void centerOrdering(multiBoxWithOrderIndex& boxes, size_t dim) {
+  if (dim == 0) {
+    // order by centerX
+    sortRuleLambdaXWithIndex comp;
+
+    std::sort(boxes.begin(), boxes.end(), comp);
+  } else {
+    // order by centerY
+    sortRuleLambdaYWithIndex comp;
+
+    std::sort(boxes.begin(), boxes.end(), comp);
+  }
+}
+
+OrderedBoxes SortInput(const std::string& onDiskBase, size_t M, uintmax_t maxBuildingRamUsage, bool workInRam) {
+  OrderedBoxes orderedInputRectangles;
+
+  ad_utility::BackgroundStxxlSorter<rTreeValue, sortRuleLambdaX> sorterRectsD0Basic = ad_utility::BackgroundStxxlSorter<rTreeValue, sortRuleLambdaX>(std::ceil((maxBuildingRamUsage < 9999999999 ? maxBuildingRamUsage : 9999999999) / 3.0));
```

Comment:
Also split everything that is only needed to build the Rtree into a separate file, then life also gets much simpler.